### DISCUSSION TITLE
For what use case is winget developed?

### DISCUSSION THREAD

**Discussion by GerhardSauer** [microsoft/winget-cli#3417] [General] [answered]
I can still see no use case for winget.
First of all it took 3 years to offer a Powershell friendly version (module) to finally return objects. This is already hilarious in itself...
And then the uninstall feature is basically utterly useless. Every software I try to uninstall with all commands available to suppress input, requires input (--silent --force --disable-interactivity, even with confirm on false in the powershell module). I need to confirm the uninstallation interactively on the graphical pop-up.
So it is impossible to uninstall without user interaction. Either there is 0 quality control or no requirements for packages to support this. 
By the way, uninstalling those software applications with their "own" uninstall commands works silent and without interaction. The the application technically supports it, just the winget uninstall process is useless.

Either way, for admins and companies, there seems to be no use case.

How do you want to address this (in the future)?

Thank you.

**Comment by Trenly** [comment, marked as answer]
winget is extremely good at saving time trying to find software when you want to install it. Instead of having to go and find the website, then run the installer, often I can just search for it on winget and install it. Winget also does especially well with portable executables; because it adds an entry for the binary into the Control Panel, it makes it much easier for me to know which portable apps I have.

Not only that, but when I got a new computer, I was able to use the export and import commands to get a majority of my applications installed on the new machine without having to go install them individually. For reference, it was over 75 different packages that were imported. Of course, it isn’t perfect, but it saved me a lot of time. Having to install 5 packages manually is a lot easier than installing 80.

That's all before the biggest use of winget - the upgrades. I remember when I first ran `winget upgrade`, there were more than 30 programs on my machine that were outdated. Some of them were outdated by several major versions. Now, these programs had been working just fine for my needs but the latest versions had a ton of bug fixes, new features, and made using them easier. Again, winget isn’t perfect - it doesn’t handle side by side installations well, and some programs don’t behave well on upgrade; but, it still works well enough that I've pinned the 3-4 packages I know have issues and the rest I let upgrade whenI run the upgrade all each week.

The uninstall flow does need some work, especially around interactivity, but there are open issues to address this. The one related to the issues you mentioned is 
* https://github.com/microsoft/winget-cli/issues/1885

Issues are given more priority the more 👍 they have, so adding your reaction on the linked issue will help.

**Reply by GerhardSauer** [reply]
You are right, I guess the upgrade function is the best part, especially, as there is an object return with the new powershell module now.
I voted for the issue, thank you.

I just want to emphasize, the tool is useful, if no user interaction is needed.
